front/pages/api/poke/plans.ts

```diff
@@ -0,0 +1,150 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getSession, getUserFromSession } from "@app/lib/auth";
+import { ReturnedAPIErrorType } from "@app/lib/error";
+import { Plan } from "@app/lib/models";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+export const PokePlanTypeSchema = t.type({
+  code: t.string,
+  name: t.string,
+  limits: t.type({
+    assistant: t.type({
+      isSlackBotAllowed: t.boolean,
+      maxMessages: t.number,
+    }),
+    connections: t.type({
+      isSlackAllowed: t.boolean,
+      isNotionAllowed: t.boolean,
+      isGoogleDriveAllowed: t.boolean,
+      isGithubAllowed: t.boolean,
+    }),
+    dataSources: t.type({
+      count: t.number,
+      documents: t.type({
+        count: t.number,
+        sizeMb: t.number,
+      }),
+    }),
+    users: t.type({
+      maxUsers: t.number,
+    }),
+  }),
+  stripeProductId: t.union([t.string, t.null]),
+  billingType: t.union([
+    t.literal("fixed"),
+    t.literal("monthly_active_users"),
+    t.literal("free"),
+  ]),
+});
+
+export type PokePlanType = t.TypeOf<typeof PokePlanTypeSchema>;
```

Comment:
cc @PopDaph this should be a top-level type called PlanType and our current PlanType should be renamed SubscribedPlanType (or SubscriptionType with a PlanType as  field which I think is the cleanest)

Fine merging as is but we want to kill PokePlanType with fire with this issue: https://github.com/dust-tt/tasks/issues/169